### ISSUE TITLE
JSON-RPC types do not agree with Ethereum's OpenRPC spec

### ISSUE THREAD

**Issue by 0xaatif** [alloy-rs/alloy#1560]
As an example, take `debug_getBadBlocks` (alphabetically, the first method in the spec)

```yaml
- name: debug_getBadBlocks
  summary: Returns an array of recent bad blocks that the client has seen on the network.
  params: []
  result:
    name: Blocks
    schema:
      title: Bad block array
      type: array
      items:
        $ref: '#/components/schemas/BadBlock'
```
https://github.com/ethereum/execution-apis/blob/v1.0.0-beta.4/src/debug/getters.yaml#L82-L91

`BadBlock` looks like
```yaml
BadBlock:
  title: Bad block
  type: object
  required:
    - block
    - hash
    - rlp
```
https://github.com/ethereum/execution-apis/blob/v1.0.0-beta.4/src/schemas/block.yaml#L140-L146

But in `alloy`, we return a `Vec<Block>`:
https://github.com/alloy-rs/alloy/blob/b621af07b7bd970d256c523c90ec0c91ec61cea1/crates/provider/src/ext/debug.rs#L30

This is the wrong struct (there's no `BadBlock` struct anyway)

**Comment by EdwardJES** [comment]
Keen to take the the `BadBlock` task.

**Comment by mattsse** [comment]
cool @EdwardJES for this we need a new helper type and then simply swap out the return type

**Comment by yash-atreya** [comment]
Closed by #1566
